File: aat-lib/src/main/java/ch/bailu/aat_lib/xml/parser/gpx/GpxBuilderInterface.java
package ch.bailu.aat_lib.xml.parser.gpx;

import org.xmlpull.v1.XmlPullParserException;

import java.io.Closeable;
import java.io.IOException;

import ch.bailu.aat_lib.gpx.interfaces.GpxPointInterface;
import ch.bailu.util_java.parser.OnParsedInterface;

/**
 * Interface to build GPX Tracks
 */
public interface GpxBuilderInterface extends Closeable, GpxPointInterface {
    void setOnRouteParsed(OnParsedInterface route);
    void setOnTrackParsed(OnParsedInterface track);
    void setOnWayParsed(OnParsedInterface way);
    void parse() throws XmlPullParserException, IOException;
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/file/xml/parser/gpx/GpxBuilderInterface.kt
package ch.bailu.aat_lib.file.xml.parser.gpx

import ch.bailu.aat_lib.gpx.interfaces.GpxPointInterface
import ch.bailu.aat_lib.file.xml.parser.util.OnParsedInterface
import org.xmlpull.v1.XmlPullParserException
import java.io.Closeable
import java.io.IOException

/**
 * Interface to build GPX Tracks
 */
interface GpxBuilderInterface : Closeable, GpxPointInterface {
    fun setOnRouteParsed(route: OnParsedInterface)
    fun setOnTrackParsed(track: OnParsedInterface)
    fun setOnWayParsed(way: OnParsedInterface)

    @Throws(XmlPullParserException::class, IOException::class)
    fun parse()
}
